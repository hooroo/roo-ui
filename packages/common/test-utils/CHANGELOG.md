# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
