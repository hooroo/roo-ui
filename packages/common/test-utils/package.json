{
  "name": "@roo-ui/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
