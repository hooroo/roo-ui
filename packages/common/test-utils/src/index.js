export { default as shallowWithTheme } from './shallowWithTheme'; // eslint-disable-line import/prefer-default-export
