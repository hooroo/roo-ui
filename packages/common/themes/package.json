{
  "name": "@roo-ui/themes",
  "version": "0.4.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
