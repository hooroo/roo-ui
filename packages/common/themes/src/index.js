import qantas from './qantas';

export { qantas }; // eslint-disable-line import/prefer-default-export
