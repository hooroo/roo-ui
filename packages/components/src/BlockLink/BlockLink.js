import { Link } from '..';

const BlockLink = Link.extend`
  display: block;
  color: inherit;
  text-decoration: none;
`;

BlockLink.displayName = 'BlockLink';

export default BlockLink;
