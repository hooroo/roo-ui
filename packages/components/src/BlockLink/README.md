# Block link

<!-- STORY -->

## Installation

```shell
$ yarn add @roo-ui/components
```

## Example

```js
import { BlockLink } from '@roo-ui/components';

export default (
  <BlockLink>Hello world</BlockLink>
);
```

### With [React Router](https://reacttraining.com/react-router/)

```js
import { Link } from 'react-router-dom';
import { BlockLink as StyledBlockLink } from '@roo-ui/components';

const BlockLink = StyledBlockLink.withComponent(Link);

export default (
  <BlockLink>Hello world</BlockLink>
);
```
