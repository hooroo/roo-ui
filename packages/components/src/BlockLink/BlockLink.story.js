import React from 'react';
import { storiesOf } from '@storybook/react';
import { withDocs } from 'storybook-readme';

import { Box } from '..';
import BlockLink from '.';
import README from './README.md';

storiesOf('Components|BlockLink', module)
  .addDecorator(withDocs(README))
  .add('default', () => (
    <BlockLink href="https://www.qantas.com" target="_blank">
      <Box p={3} bg="grey.6">
        Hello world
      </Box>
    </BlockLink>
  ));
