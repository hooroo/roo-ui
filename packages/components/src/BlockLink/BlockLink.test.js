import React from 'react';
import { qantas as theme } from '@roo-ui/themes';
import { shallowWithTheme } from '@roo-ui/test-utils';

import BlockLink from '.';

describe('<BlockLink />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowWithTheme(<BlockLink>Hello world</BlockLink>, theme);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
