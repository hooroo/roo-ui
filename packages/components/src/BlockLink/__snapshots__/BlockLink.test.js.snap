// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockLink /> renders correctly 1`] = `
.c0 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  color: #E00E00;
  display: block;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  color: #870000;
}

.c0:focus {
  outline: 2px solid #8DE2E0;
}

<Clean.a
  blacklist={
    Array [
      "color",
      "bg",
      "hover",
      "fontWeight",
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "hidden",
      "underline",
    ]
  }
  className="c0"
  hidden={false}
  underline={false}
>
  Hello world
</Clean.a>
`;
